crates/bevy_core/src/time/timer.rs

```diff
@@ -4,39 +4,56 @@ use bevy_property::Properties;
 use std::time::Duration;
 
 /// Tracks elapsed time. Enters the finished state once `duration` is reached.
+///
+/// Non repeating timers will stop tracking and stay in the finished state until reset.
+/// Repeating timers will only be in the finished state on each tick `duration` is reached or exceeded, and can still be reset at any given point.
 #[derive(Clone, Debug, Default, Properties)]
 pub struct Timer {
     pub elapsed: f32,
     pub duration: f32,
     pub finished: bool,
+    prev_finished: bool,
```

Comment:
what this field doesn't isn't imediatly obvious, could you document it?

Reply:
Added some comments to `prev_finished` and `just_finished()` to clarify.